app/controllers/bacons_controller.rb

```diff
@@ -56,6 +66,10 @@ def stats
     ]
   end
 
+  def versions
+    @versions ||= JSON.parse(params[:versions]) rescue Hash.new('unknown')
```

Comment:
Yep, this returns an empty Hash as the value of the constructor, but then the instance returns `'unknown'` instead of `nil` if you ask for a key that has no value